plugins/by-name/fidget/default.nix

```diff
@@ -34,650 +28,424 @@ let
-          option, or manually calling `|fidget.notification.reset|`.
+      display = {
+        render_limit = defaultNullOpts.mkNullableWithRaw (with types; either number (enum [ false ])) 16 ''
+          How many LSP messages to show at once.
 
-          Set to `false` to disable polling altogether; you can still manually poll progress
-          messages by calling `|fidget.progress.poll|`.
+          If set to `false`, no limit will be enforced.
         '';
 
-        suppressOnInsert = helpers.defaultNullOpts.mkBool false ''
-          Suppress new messages while in insert mode.
+        done_ttl = defaultNullOpts.mkNum 3 ''
+          How long a message should persist after completion.
+        '';
 
-          Note that progress messages for new tasks will be dropped, but existing tasks will be
-          processed to completion.
+        done_icon = mkIconOption "✔" ''
+          Icon shown when all LSP progress tasks are complete.
         '';
 
-        ignoreDoneAlready = helpers.defaultNullOpts.mkBool false ''
-          Ignore new tasks that are already complete.
+        done_style = defaultNullOpts.mkStr "Constant" ''
+          Highlight group for completed LSP tasks.
+        '';
 
-          This is useful if you want to avoid excessively bouncy behavior, and only seeing
-          notifications for long-running tasks. Works best when combined with a low `pollRate`.
+        progress_ttl = defaultNullOpts.mkNum { __raw = "math.huge"; } ''
+          How long a message should persist when in progress.
         '';
 
-        ignoreEmptyMessage = helpers.defaultNullOpts.mkBool true ''
-          Ignore new tasks that don't contain a message.
+        progress_icon = mkIconOption [ "dots" ] ''
+          Icon shown when an LSP progress task is in progress.
+        '';
 
-          Some servers may send empty messages for tasks that don't actually exist.
-          And if those tasks are never completed, they will become stale in Fidget.
-          This option tells Fidget to ignore such messages unless the LSP server has anything
-          meaningful to say.
+        progress_style = defaultNullOpts.mkStr "WarningMsg" ''
+          Highlight group for in-progress LSP tasks.
+        '';
 
-          Note that progress messages for new empty tasks will be dropped, but existing tasks will
-          be processed to completion.
+        group_style = defaultNullOpts.mkStr "Title" ''
+          Highlight group for group name (LSP server name).
         '';
 
-        notificationGroup = helpers.defaultNullOpts.mkLuaFn "function(msg) return msg.lsp_name end" ''
-          How to get a progress message's notification group key
+        icon_style = defaultNullOpts.mkStr "Question" ''
+          Highlight group for group icons.
+        '';
 
-          Set this to return a constant to group all LSP progress messages together.
+        priority = defaultNullOpts.mkNullableWithRaw (with types; either number (enum [ false ])) 30 ''
```

Comment:
This didn't support false before, is it similar to the other ones that support it for disabling some functionality? We should probably update the description to match the others, if thats the case.

Reply:
I guess that it is. But it is don't know exactly what `false` does here exactly, since it doesn't have anything in the plugin documentation. I will try to figure that out.

https://github.com/j-hui/fidget.nvim/blob/9238947645ce17d96f30842e61ba81147185b657/lua/fidget/progress/display.lua#L100

Reply:
Hmm, I guess they didn't document it.. we can leave it without explanation, until someone wants to fill in the information.